api/src/main/java/org/pmiops/workbench/api/CohortController.java

```diff
@@ -1,29 +1,174 @@
 package org.pmiops.workbench.api;
 
-import com.google.common.collect.ImmutableList;
-import java.util.List;
+import java.sql.Timestamp;
+import java.time.Clock;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import javax.inject.Provider;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.pmiops.workbench.db.dao.CohortDao;
+import org.pmiops.workbench.db.dao.WorkspaceDao;
+import org.pmiops.workbench.db.model.User;
+import org.pmiops.workbench.db.model.Workspace;
+import org.pmiops.workbench.exceptions.NotFoundException;
 import org.pmiops.workbench.model.Cohort;
+import org.pmiops.workbench.model.CohortListResponse;
+import org.pmiops.workbench.model.DataAccessLevel;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.RestController;
 
 @RestController
-public class CohortController implements CohortsApi {
+public class CohortController implements CohortsApiDelegate {
+
+  private static final Logger log = Logger.getLogger(CohortController.class.getName());
+
+  private static final Function<org.pmiops.workbench.db.model.Cohort, Cohort> TO_CLIENT_COHORT =
+      new Function<org.pmiops.workbench.db.model.Cohort, Cohort>() {
+        @Override
+        public Cohort apply(org.pmiops.workbench.db.model.Cohort cohort) {
+          Cohort result = new Cohort();
+          result.setLastModifiedTime(new DateTime(cohort.getLastModifiedTime(), DateTimeZone.UTC));
+          if (cohort.getCreator() != null) {
+            result.setCreator(cohort.getCreator().getEmail());
+          }
+          result.setCriteria(cohort.getCriteria());
+          result.setCreationTime(new DateTime(cohort.getCreationTime(), DateTimeZone.UTC));
+          result.setDescription(cohort.getDescription());
+          result.setId(cohort.getExternalId());
+          result.setName(cohort.getName());
+          result.setType(cohort.getType());
+          return result;
+        }
+      };
+
+  private static final Function<Cohort, org.pmiops.workbench.db.model.Cohort> FROM_CLIENT_COHORT =
+      new Function<Cohort, org.pmiops.workbench.db.model.Cohort>() {
+        @Override
+        public org.pmiops.workbench.db.model.Cohort apply(Cohort cohort) {
+          org.pmiops.workbench.db.model.Cohort result = new org.pmiops.workbench.db.model.Cohort();
```

Comment:
It looks like you can either use `setFoo` which returns void, or `foo()` which is chainable; not sure why there are both. So in cases like this I might rather
`return result.criteria(...).description(...).externalId(..);` to avoid repeating the variable name.